core/templates/pages/skill-editor-page/editor-tab/skill-misconceptions-editor/skill-misconceptions-editor.directive.ts

```diff
@@ -161,10 +163,17 @@ angular.module('oppia').directive('skillMisconceptionsEditor', [
           ctrl.$onInit = function() {
             $scope.skill = SkillEditorStateService.getSkill();
             $scope.misconceptions = $scope.skill.getMisconceptions();
-            $scope.$on(EVENT_SKILL_REINITIALIZED, function() {
-              $scope.misconceptions = $scope.skill.getMisconceptions();
-            });
+            ctrl.parentSubscription.add(
+              SkillEditorStateService.getSkillReinitializedSubject()
```

Comment:
Reduce the number of like whenever possible. The preferred way is to break after `(`. Please see my suggestions below.

Reply:
Done!